### PR TITLE
fix: provider requests signup data and redirectTo

### PR DESCRIPTION
**Pull request by elitan**
This PR fixes:

- Add signup data to request (like metadata, roles, displayName etc)
- Use correct variable name for redirecting: `redirectTo` instead of `redirectUrl`

fixes: https://github.com/nhost/hasura-auth/issues/80, https://github.com/nhost/nhost/issues/188

### CODE REVIEW

File: migrations/00004_provider-request-generic.sql [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+BEGIN;
+
+ALTER TABLE
+  auth.provider_requests
+DROP COLUMN redirect_url;
+
+ALTER TABLE
+  auth.provider_requests
+ADD
+  COLUMN metadata JSONB;
```
Comment (elitan) [inline migrations/00004_provider-request-generic.sql:11]: Either we use a general `metadata JSONB` column or create individual columns for:

- redirectTo
- displayName
- locale
- defaultRole
- allowedRoles
- metadata

Comment (elitan) [reply migrations/00004_provider-request-generic.sql:11]: It's also safe to do this change later if we want to.

Comment (plmercereau) [reply migrations/00004_provider-request-generic.sql:11]: I would rather name the column options than metadata as it captures all registration options, including options.metadata, options.redirectTo etc

**Review by plmercereau — APPROVED** [review]

**Review by nunopato — COMMENTED** [review]
We spoke about this offline, but leaving my comments here as well. 

Since `redirectTo` makes sense to be a `signIn` option, we should have it per provider request, hence be a column. 

The other options are `signUp` options, meaning they should only be set the first time the user uses a provider and only for context (the data is also stored in users). A JSON blob field seems the right choice.

**Review by nunopato — APPROVED** [review]
lgtm

**Pull request merged**